infrastructure/src/main/java/org/corfudb/infrastructure/log/SegmentUtils.java

```diff
@@ -0,0 +1,170 @@
+package org.corfudb.infrastructure.log;
+
+import com.google.common.base.Preconditions;
+import com.google.protobuf.AbstractMessage;
+import com.google.protobuf.ByteString;
+import io.netty.buffer.Unpooled;
+import org.corfudb.common.compression.Codec;
+import org.corfudb.protocols.logprotocol.CheckpointEntry;
+import org.corfudb.protocols.wireprotocol.LogData;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+public final class SegmentUtils {
```

Comment:
nit (same as above): 

> Can this class be refactored to this location? https://github.com/CorfuDB/CorfuDB/blob/4955f4e439c22a2966176a61c530786fa438fc62/infrastructure/src/main/java/org/corfudb/infrastructure/utils

Reply:
This file mostly contains type definitions and not general enough to go into `Persistence.java`

Reply:
Sorry, I was talking about moving the class `SegmentUtils` to the `org/corfudb/infrastructure/utils` package, not merging it into the `Persistence.java` class.

Reply:
Done.